lib/ansiblelint/errors.py

```diff
@@ -18,20 +22,52 @@ def __repr__(self):
         return formatstr.format(self.rule.id, self.message,
```

Comment:
How come `self.rule.id` is expected to always exist here but not in other places?

Reply:
One of the reasons why I proposed https://github.com/ansible/ansible-lint/pull/884